### ISSUE TITLE
Blocksize Storage Options

### ISSUE THREAD

**Issue by VinnyPembe** [greenplum-db/gpdb#2025]
Hello,

I have recently been doing a lot of system performance testing in the way of SSDs and RAID configuration options. I've concluded, as is already known amongst many, that a 256KB blocksize performs better for random reads than a 32KB blocksize.

Now, I understand that you can set the blocksize specifically (and only) for Append-Only, Column-Orientated tables.. Personally, this is not useful for me at all (unfortunately) and aside testing that it does indeed perform much better (which it does!), I cannot use it in practice the majority of the time.

What I am wanting to ask/know is if at any point blocksize options may be available for Heap tables (or database settings in general on initialisation) to allow for this performance gain to be more useable?

Regards,
Vinny Pemberton.

**Comment by hlinnaka** [comment]
It would be quite difficult to increase the heap block size further, from 32kB, because there are many places throughout the code that assume that an offset into a block can be stored in a 16-bit integer.

I find it hard to believe that there would be a significant performance gain to be had from that anywya, to be honest, especially for random reads. How did you reach that conclusion?

**Comment by VinnyPembe** [comment]
Ah, that is also what the Pivotal Engineer I spoke to said - I was hoping (though highly unlikely it was) that perhaps the GitHub version may have been working towards it in some way.

Significant - Perhaps not quite so literally. Noticable - Yes. It also, of course, depends on the performance you have with disks in the first place. To gain an extra 500KB/s performance on a 4MB/s disk is pointless, but to gain 500MB/s on a 4GB/s is quite a nice welcomed gain, and can make a noticable difference (especially given you're utilising what you already have anyway)!

I measured disk performance (IO per secs) using the FIO tool at 32, 64 and 256 KB blocksizes (with various RAID 5 setups) and found 256 to be the best performing. I also concluded this with the Pivotal Engineer I spoke with, as Pivotal themselves recommend using a 256KB blocksize setup where possible.

I still have yet to run a thorough TPC-H Benchmark test against this particular setup, though I did a brief small test and it was indeed faster on the 256.
